Increases 'nofile' ulimit value in init scripts.

Yggdrasil opened the pull request:
The default maximum number of open files on many systems (1024) is not sufficient for Traefik which maintains many open sockets. Increasing this to 16384 should be a more reasonable default and should cause no trouble (AFAIK).

JayH5 commented:
Thanks :)

Not sure how many times this JSON gem issue will break.. let me try fix it in `develop`.

JayH5 commented:
@Yggdrasil thanks for the PR.. I would prefer if the ulimit number were configurable through a config parameter, though.

Yggdrasil commented:
Good point. 😄
I've parameterized the value, including validation. Also rebased.
Tested on Ubuntu 14.04. Haven't tested the Systemd init, but the setting should work according to the docs.

JayH5 commented:
Thanks. This will still need tests to ensure the value is actually written to the init file.. similar to what we do for the binary location here: https://github.com/praekeltfoundation/puppet-traefik/blob/develop/spec/classes/install_spec.rb#L62

Yggdrasil commented:
I'm a beginner at testing Puppet modules so this was a bit daunting. 😅 

I removed the `include stdlib` line and added tests (thanks for the example!) for both the default value and a custom value.

I've had to increase the max line width in rubocop, so I changed it from the default of 80 to 140, which is recommended in the Puppet style guide. Hope that's ok.

Not sure how to get rid of the 'Changes requested' error above.

Also, the code coverage warns:
```
Untouched resources:
  Class[Traefik]
  Traefik::Config::File_rule[test]
  Traefik::Config::Section[test]
```
Perhaps you can take a look at that?

JayH5 commented on .rubocop.yml:
```diff
@@ -18,3 +18,7 @@ Style/SpaceInsideHashLiteralBraces:
 Style/WordArray:
   Enabled: false
+
+# Inclease linelentgh to match Puppet style guide
```
Can you fix the typos here and add a newline at the end of the file, please?

Yggdrasil replied:
Done!

JayH5 commented:
@Yggdrasil thanks for putting in the effort with the tests. Most people give up when I ask for Puppet tests 😅. It looks good :)

> I've had to increase the max line width in rubocop, so I changed it from the default of 80 to 140, which is recommended in the Puppet style guide. Hope that's ok.

:+1:

> Not sure how to get rid of the 'Changes requested' error above.

That goes away when I mark my code review as "Approved" in Github.

> Also, the code coverage warns:

Don't worry about that. The code coverage stuff in `rspec-puppet` is super rudimentary. If you create a resource anywhere then it needs to have an assertion on it or it's not considered "covered". But in a lot of cases we create resources to test other things and not the test resources themselves...if that makes sense? Also, in order to assert on the parent class (`Class[Traefik]`) we'd need to do proper integration tests which I haven't figured out how to nicely do yet..

There's just one more change I'd like here and then we should be good to go.

JayH5 commented:
Thank you :cake:
